conformance/lang/expressions/error_constructor.balt

```diff
@@ -0,0 +1,526 @@
+
+function init() {
+    error err1 = error("error message 1");
+    io:println(err1.detail() is map<value:Cloneable>); // @output true
+    io:println(err1.detail() is readonly); // @output true
+    io:println(err1.detail()); // @output {}
+
+    error err = error("ERROR!");
+    error err2 = error("error message 2", err);
+    io:println(err2.detail() is map<value:Cloneable>); // @output true
+    io:println(err2.detail() is readonly); // @output true
+    io:println(err2.detail()); // @output {}
+
+    map<string> m = {"x": "x", "y": "y"};
+    error err3 = error("error message 3", a = "a", b = false, c = 1, d = 1.3, e = err, f = [1, 2, 3], g = m);
+    io:println(err3.detail() is map<value:Cloneable>); // @output true
+    io:println(err3.detail() is readonly); // @output true
+    io:println(err3.detail()); // @output {"a":"a","b":false,"c":1,"d":1.3,"e":error("ERROR!"),"f":[1,2,3],"g":{"x":"x","y":"y"}}
+
+    error err4 = error("error message 4", err, a = "a", b = false, c = 1, d = 1.3, e = err, f = [1, 2, 3], g = m);
+    io:println(err4.detail() is map<value:Cloneable>); // @output true
+    io:println(err4.detail() is readonly); // @output true
+    io:println(err4.detail()); // @output {"a":"a","b":false,"c":1,"d":1.3,"e":error("ERROR!"),"f":[1,2,3],"g":{"x":"x","y":"y"}}
+
+    error err5 = error E("error message 1", message = "Oops!");
+    io:println(err5.detail() is map<value:Cloneable>); // @output true
+    io:println(err5.detail() is readonly); // @output true
+    io:println(err5.detail()); // @output {"message":"Oops!"}
+
+    error err6 = error E("error message 2", err, message = "Oops!", fatal = true);
+    io:println(err6.detail() is readonly); // @output true
+    io:println(err6.detail() is map<value:Cloneable>); // @output true
+    io:println(err6.detail()); // @output {"message":"Oops!","fatal":true}
+}
+
+Test-Case: error
+Description: Test the static type of each named-arg must be a subtype of value:Cloneable.
+Labels: error-constructor-expr, module-class-defn, error-type, explicit-new-expr, string, value:Cloneable
+
+class ErrorClass {
+
+    private error e;
+
+    public function init() {
+        self.e = error("ERROR!", fatal = false);
+    }
+
+    public function get() returns error {
+        return self.e;
+    }
+}
+
+function errorFunction() {
+    ErrorClass err = new ErrorClass();
+    error err1 = error("error message 3", err = m); // @error invalid arg type in error detail field 'err', expected 'ballerina/lang.value:0.0.0:Cloneable', found 'ErrorClass'
+}
+
+Test-Case: error
+Description: Test the arg-name of every named-arg must be specified as the field-name of an individual-field-descriptor occurring in type descriptor D of the detail mapping.
+Labels: error-constructor-expr, module-type-defn, record-type, error-type, string
+
+type E1 error<record {string message?;}>;
+
+type E2 error<record {|string message?;|}>;
+
+type E3 error<record {||}>;
+
```

Comment:
We can also test when the record has required fields?
And also with an exclusive record type having a rest field?